apollo-federation/src/query_graph/path_tree.rs

```diff
@@ -365,8 +369,18 @@ where
                         (Some(cond_a), Some(cond_b)) => cond_a.equals_same_root(cond_b),
                         _ => false,
                     }
-                    && a.context_to_selection == b.context_to_selection
```

Comment:
Bugs:
- `None` should be considered the same as an empty collection, similar to JS. (Really we should probably remove the `Option` wrapper in the data structures, but will leave for a follow-up.)

Reply:
There gist is that in this case, there's no distinction, and we should remove the `Option` (both for matching context IDs, and for maps of arguments to context usages).